lib/fx/adapters/postgres/functions.rb

```diff
@@ -17,9 +17,22 @@ class Functions
               ON pn.oid = pp.pronamespace
           LEFT JOIN pg_depend pd
               ON pd.objid = pp.oid AND pd.deptype = 'e'
-          WHERE pn.nspname = 'public' AND pd.objid IS NULL
+          WHERE pn.nspname = 'public' AND pd.objid IS NULL AND pp.proisagg = 'f'
           ORDER BY pp.oid;
-        EOS
+        SQL
+
+        PG11_FUNCTIONS_WITH_DEFINITIONS_QUERY = <<-SQL.squish.freeze
```

Comment:
I'd ideally like to have 1 query for both PG10 and 11. Is that doable?

Reply:
I think there are 2 possible ways:

1. Move this to interpolated string instead of constant.
2. Keep main part of query in constant and concat it with conditional part dynamically.

Reply:
Neither of those sound like a good idea to me.

I'll re-iterate. Is it possible to have 1 query that returns the functions with their definitions on Postgres 10 and 11?

Reply:
I added temp function, now it's possible.

Reply:
This still doesn't feel right, and I don't feel that my initial question has been answered.

Is there a way to get the functions, with their definitions, that's portable between PG 10 and 11? If it isn't, I don't mind having two functions. But, ideally I'd like there to be only one. But - if that's possible, so be it.

Having a functions feels even more awkward.

Reply:
@teoljungberg Ah, no, it's impossible.

Reply:
Is there any documentation or a changelog between PG 10 and 11 that supports this claim?

Reply:
https://www.postgresql.org/docs/release/11.0/

> Replace system catalog pg_proc's proisagg and proiswindow columns with prokind (Peter Eisentraut)

This new column more clearly distinguishes functions, procedures, aggregates, and window functions.

Reply:
Good to know!

If you can document this in the PR description, and split this up to use two different queries I think we're good here.

Reply:
Sure, I'll make changes soon

Reply:
@teoljungberg Done :)